src/webpack/middlewares/OriginSubstitution.js

```diff
@@ -0,0 +1,51 @@
+const debug = require('../../util/debug').here(__filename);
+const url = require('url');
+const harmon = require('harmon');
+const through = require('through');
+const removeTrailingSlash = x => x.replace(/\/$/, '');
+module.exports = function createOriginSubstitutionMiddleware(
+    oldDomain,
+    newDomain
+) {
+    const oldOrigin = removeTrailingSlash(url.format(oldDomain));
+    const newOrigin = removeTrailingSlash(url.format(newDomain));
+    const attributesToReplaceOrigin = ['href', 'src', 'style'].map(attr => ({
+        query: `[${attr}*="${oldOrigin}"]`,
+        func(node) {
+            node.setAttribute(
+                attr,
+                node
+                    .getAttribute(attr)
+                    .split(oldOrigin)
+                    .join(newOrigin)
+            );
+        }
+    }));
+    const tagsToReplaceOrigin = ['style'].map(attr => ({
+        query: attr,
+        func(node) {
+            const stream = node.createStream();
```

Comment:
Do we have to work with a stream here? We're not getting any benefit from it because we're calling `toString` so the whole thing gets buffered into memory

Reply:
We call `toString` on a chunk of buffer, not the whole stream. And yeah, this is a module that uses substack/trumpet under the hood, so streams are the only interface.